File: java/src/main/java/com/dddeurope/niuz/website/Teaser.java
package com.dddeurope.niuz.website;

import java.util.Objects;

public class Teaser {
    private final String headline;
    private final String author;

    public Teaser(String headline, String author) {
        this.headline = headline;
        this.author = author;
    }

    public String getHeadline() {
        return headline;
    }

    public String getAuthor() {
        return author;
    }

    @Override
    public String toString() {
        return "Teaser{" +
                "headline='" + headline + '\'' +
                ", author='" + author + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Teaser teaser = (Teaser) o;
        return Objects.equals(headline, teaser.headline) && Objects.equals(author, teaser.author);
    }

    @Override
    public int hashCode() {
        return Objects.hash(headline, author);
    }
}

File: dotnet/niuz.application/website/Teaser.cs
using System;

namespace niuz.application.website
{
    public class Teaser
    {
        public string Headline { get; }

        public string Author { get; }

        public Teaser(string headline, string author)
        {
            Headline = headline;
            Author = author;
        }

        protected bool Equals(Teaser other)
        {
            return Headline == other.Headline && Author == other.Author;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Teaser) obj);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Headline, Author);
        }
    }
}